src/keysignature.ts

```diff
@@ -114,11 +124,11 @@ export class KeySignature extends StaveModifier {
     }
 
     // Place the glyph on the stave
-    this.placeGlyphOnLine(glyph, this.stave, acc.line);
+    this.placeGlyphOnLine(glyph, this.checkStave(), acc.line);
     this.glyphs.push(glyph);
 
     const xPosition = this.xPositions[this.xPositions.length - 1];
-    const glyphWidth = glyph.getMetrics().width + extraWidth;
+    const glyphWidth = glyph.getMetrics().width ?? 0 + extraWidth;
```

Comment:
Can we remove the `?? 0` -- there could be a silent failure here.

Reply:
done